tfe/resource_tfe_agent_pool.go

```diff
@@ -109,3 +110,27 @@ func resourceTFEAgentPoolDelete(d *schema.ResourceData, meta interface{}) error
 
 	return nil
 }
+
+func resourceTFEAgentPoolImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	tfeClient := meta.(*tfe.Client)
+
+	s := strings.Split(d.Id(), "/")
+	if len(s) >= 3 {
+		return nil, fmt.Errorf(
+			"invalid agent pool input format: %s (expected <ORGANIZATION>/<AGENT POOL NAME> or <AGENT POOL ID>)",
+			d.Id(),
+		)
+	} else if len(s) == 2 {
+		org := s[0]
+		poolName := s[1]
+		poolID, err := fetchtAgentPoolID(org, poolName, tfeClient)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"error retrieving agent pool with name %s from organization %s %w", poolName, org, err)
+		}
+
+		d.SetId(poolID)
+	}
```

Comment:
should we handle cases where users accidentally throw an extra slash or two in there? something like `/my-org/my-agent-pool/`

Reply:
That's an interesting questions.... if they put slashes anywhere other than the front or end (eg `my/org/my-agent-pool`) it will error out and tell the user the correct format

I guess if they add extra slashes that dont break our parsing...do we care? Im not really sure what the correct behaviour there should even be

Reply:
Currently all custom importer funcs will error if the import ID doesn't follow the exact format specified. To maintain consistency, we should leave it as is :+1: